### PR TITLE
Support @PermissionsAllowed defined on meta-annotation

### LINKED ISSUE #39663: Security - Ability to create custom permission annotation
### Description

Inspired by a conversation on Zulip: https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/.E2.9C.94.20grpc.20-.20current.20grpc.20message.20in.20auth, I propose a new feature for custom permission annotation for gRPC and REST.

Concept:

Allow creation of custom permission annotations, themselves annotated with @PermissionsAllowed.
You would be able to define default permission values within these annotations and other values as like for example value itself.
Optionally, it would be nice to have passing custom parameters to the custom permission class from custom permission (Explained in implementation), if specified.

### Implementation ideas

As suggested in conversation you would be able to do this:
```java
@PermissionsAllowed(permission = CustomPermission.class, value = "request")
public @interface CustomPermissionAnnotation {

}
```

Or you would be able to do this:
```java
@PermissionsAllowed(permission = CustomPermission.class, value = "request")
public @interface CustomPermissionAnnotation {
    CustomEnumType customEnumType() default CustomEnumType.READ;
}

public class CustomPermission extends Permission {

    private final String name;
    private final String[] actions;
    private final CustomEnumType customType;

    public CustomPermission(String name, String[] actions, CustomEnumType customEnumType) {
    
    }
```

You would be then able to do this:
```java
@CustomPermissionAnnotation(customEnumType = CustomEnumType.WRITE)
public String writeCustomPermission(String write) {
    
}
```

### PR DESCRIPTION
**Pull request by michalvavrik**
- Closes: https://github.com/quarkusio/quarkus/issues/39663

### CODE REVIEW

File: docs/src/main/asciidoc/security-authorize-web-endpoints-reference.adoc [hunk 1]
```diff
@@ -1054,6 +1054,29 @@ Because `MediaLibrary` is the `TvLibrary` class parent, a user with the `admin`
 
 CAUTION: Annotation-based permissions do not work with custom xref:security-customization.adoc#jaxrs-security-context[Jakarta REST SecurityContexts] because there are no permissions in `jakarta.ws.rs.core.SecurityContext`.
 
+[[permission-meta-annotation]]
+=== Create permission meta-annotations
+
+`@PermissionsAllowed` can also be used in meta-annotations. Example below creates new `@CanWrite` security annotation:
```
Comment (sberyozkin) [inline docs/src/main/asciidoc/security-authorize-web-endpoints-reference.adoc:1060]:
```suggestion
`@PermissionsAllowed` can also be used in meta-annotations.
For example, a new `@CanWrite` security annotation can be created like this:
```

Suggesting a new line as per Doc team guidelines, with a minor rewording

**Review by sberyozkin — APPROVED** [review]

File: docs/src/main/asciidoc/security-authorize-web-endpoints-reference.adoc (hunk 1 above)
Comment (michalvavrik) [reply docs/src/main/asciidoc/security-authorize-web-endpoints-reference.adoc]: Done, thanks.

**Comment by michalvavrik** [comment]
FYI @sberyozkin there is some WebAuthn failure that is not related:

```
2024-09-12T18:29:00.3799005Z Caused by: java.lang.RuntimeException: org.testcontainers.containers.ContainerFetchException: Can't get Docker image: RemoteDockerImage(imageName=docker.io/postgres:14, imagePullPolicy=DefaultPullPolicy(), imageNameSubstitutor=org.testcontainers.utility.ImageNameSubstitutor$LogWrappedImageNameSubstitutor@202902d5)
2024-09-12T18:29:00.3802884Z 	at io.quarkus.datasource.deployment.devservices.DevServicesDatasourceProcessor.startDevDb(DevServicesDatasourceProcessor.java:367)
2024-09-12T18:29:00.3805407Z 	at io.quarkus.datasource.deployment.devservices.DevServicesDatasourceProcessor.launchDatabases(DevServicesDatasourceProcessor.java:121)
2024-09-12T18:29:00.3807386Z 	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:732)
2024-09-12T18:29:00.3808752Z 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
2024-09-12T18:29:00.3809867Z 	at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
2024-09-12T18:29:00.3810901Z 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
2024-09-12T18:29:00.3812163Z 	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
2024-09-12T18:29:00.3813567Z 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
2024-09-12T18:29:00.3814945Z 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
2024-09-12T18:29:00.3816056Z 	at java.base/java.lang.Thread.run(Thread.java:840)
2024-09-12T18:29:00.3816886Z 	at org.jboss.threads.JBossThread.run(JBossThread.java:483)
```

Let me know if you require rebase on current main. (well, we will see how other in progress runs go) You could also re-trigger just this particular run in the `checks` tab.

**Comment by michalvavrik** [comment]
hey @geoand , I remember you triggered just one failed run in CI in one of my PRs. Could you please retrigger just `integration-tests/security-webauthn`? It failed over pulling of Docker image. Don't want to run whole CI as it takes long in this PR.

**Comment by geoand** [comment]
Done!

**Comment by michalvavrik** [comment]
> Done!

Thanks!

**Pull request merged**